src/daft-core/src/array/ops/utf8.rs

```diff
@@ -159,6 +159,103 @@ impl Utf8Array {
         }
     }
 
+    pub fn extract(&self, pattern: &Utf8Array, index: usize) -> DaftResult<Utf8Array> {
+        let self_arrow = self.as_arrow();
+        let pattern_arrow = pattern.as_arrow();
+        // Handle all-null cases.
+        if self_arrow
+            .validity()
+            .map_or(false, |v| v.unset_bits() == v.len())
+            || pattern_arrow
+                .validity()
+                .map_or(false, |v| v.unset_bits() == v.len())
+        {
+            return Ok(Utf8Array::full_null(
+                self.name(),
+                self.data_type(),
+                std::cmp::max(self.len(), pattern.len()),
+            ));
```

Comment:
This is an unusual edge case but what if one of the validity maps was all false but `self` and `pattern` had different, non-unit lengths?

Reply:
yeah it would just return a series of nulls with length = `max(self.len, pattern.len)`, its the same behavior we have for str.split rn: https://github.com/Eventual-Inc/Daft/blob/main/src/daft-core/src/array/ops/utf8.rs#L107-L118